src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -36,6 +36,9 @@ internal static class ThreadPoolGlobals
         public static volatile bool vmTpInitialized;
         public static bool enableWorkerTracking;
 
+        public const int wsqNowActive = 2; // Require at least 2 full observations of empty wsq to transition to inactive
+        public static int wsqActive;
```

Comment:
If there were multiple appdomains, it would only steal across its own appdomain's threads since it wouldn't be able to see statics of a different appdomain. I believe thread-locals are also separated.